Lesson27/aysel/update.js

```diff
@@ -0,0 +1,51 @@
+const urlParams = new URLSearchParams(window.location.search);
+const postId = urlParams.get('postId');
+
+//  TODO: 
+// 1. Fetch post data for the post with given postId
+// 2. Prefill form with post title and post body
+// 3. Select form element and attache 'submit' event listener to it, send new post information with UPDATE request
+// 4. Show confirmation to the user if update operation was successful
+
+
+document.addEventListener('DOMContentLoaded', function() {
+    if (postId) {
+        getPostById(postId).then(postData => {
+            document.getElementById('title').value = postData.title;
+            document.getElementById('body').value = postData.body;
+        }).catch(error => console.error('Error fetching post:', error)); 
+    }
+
+    document.getElementById('update-post-form').addEventListener('submit', function(event) {
+        event.preventDefault();
+        updatePost(postId); 
+    });
+});
+
+function getPostById(id) {
+    return fetch(`https://jsonplaceholder.typicode.com/posts/${id}`)
+    .then(response => response.json())
+    .catch(error => console.error('Error fetching post by ID:', error)); 
+}
+
+function updatePost(id) { 
+    const updatedPost = {
+        title: document.getElementById('title').value,
+        body: document.getElementById('body').value,
+        userId: 1
+    };
+
+    fetch(`https://jsonplaceholder.typicode.com/posts/${id}`, {
+        method: 'PUT',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(updatedPost)
+    })
+    .then(response => response.json())
+    .then(data => {
+        console.log('Post updated:', data);
+        window.location.href = 'index.html'; 
```

Comment:
It's better to delay the redirect a bit. 
E.g. you can show a confirmation message on the screen first for 3-4 seconds (just add a div element in HTML and change it's `innerText` to a confirmation message, event better to add the title and body of the post that user typed in the form.) Then, using `setTimeout` https://developer.mozilla.org/en-US/docs/Web/API/setTimeout
function redirect the user to the main page.

Alternatively, you can skip using `setTimeout` and redirection. Just add confirmation message to the screen as described above and add a 'go back' button (<a> element) that will bring the user to the main page.